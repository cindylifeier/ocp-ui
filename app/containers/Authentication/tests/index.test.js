import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'mock-local-storage';

import { Authentication } from '../index';

configure({ adapter: new Adapter() });

describe('<Authentication />', () => {
  it('should match snapshot', () => {
    // Arrange
    const isAuthenticated = true;
    const auth = {
      isAuthenticated,
    };
    const location = '/test-location';
    const props = {
      auth,
      location,
    };

    // Act
    const renderedComponent = shallow(<Authentication {...props} />);

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
});
