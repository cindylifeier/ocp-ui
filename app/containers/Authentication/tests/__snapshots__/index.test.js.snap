// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authentication /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Authentication
    auth={
      Object {
        "isAuthenticated": true,
      }
    }
    location="/test-location"
    workspace={
      Object {
        "workflowRole": "Patient",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "push": false,
      "to": Object {
        "pathname": "/ocp-ui/login",
        "state": Object {
          "from": "/test-location",
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "push": false,
        "to": Object {
          "pathname": "/ocp-ui/login",
          "state": Object {
            "from": "/test-location",
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
